### PR TITLE
No Grid sorting when holding shift. 

### PR DESCRIPTION
**Pull request by Darkere**
https://gfycat.com/FlusteredRecklessChafer

Happens when holding shift and cursor is over grid slots or over the crafting grids output slot.

### CODE REVIEW

File: src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java [hunk 1]
```diff
@@ -559,4 +576,8 @@ public void updateScrollbar() {
     public static IGridSorter getDefaultSorter() {
         return new NameGridSorter();
     }
+
+    public boolean avoidSorting() {
+        return hasShiftDown() && !doSort;
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java]: Can this `hasShiftDown` check be moved to the place where `doSort` is modified?

File: src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java [hunk 2]
```diff
@@ -81,7 +82,7 @@ public void onPageChanged(int page) {
     @Override
     protected void onPreInit() {
         super.onPreInit();
-
+        doSort = true;
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java]: this.doSort

File: src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java (hunk 1 above)
Comment (raoulvdberge) [reply src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java]: And perhaps rename this to `canSort`.

File: src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java [hunk 3]
```diff
@@ -55,6 +55,7 @@
     private TabListWidget tabs;
 
     private boolean wasConnected;
+    boolean doSort;
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java]: `private boolean`

**Review by raoulvdberge — CHANGES_REQUESTED** [review]
Please also keep in mind the formatting. We use standard IntellIJ formatting rules.

File: src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java (hunk 1 above)
Comment (Darkere) [reply src/main/java/com/raoulvdberge/refinedstorage/screen/grid/GridScreen.java]: As `doSort` gets modified in `onMouseClick` moving `hasShiftDown` there introduces multiple edge cases.

**Comment by raoulvdberge** [comment]
Thx!

**Pull request merged**